(baby-project) adds CI support

dusan-turajlic opened the pull request.

dusan-turajlic committed:
(baby-project) adds CI support

AndreasNasman commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,26 @@
+
+on:
+  push:
+    branches: [ master ]
```
```suggestion
    branches: [ main ]
```

AndreasNasman commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,26 @@
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
```suggestion
    branches: [ main ]
```

AndreasNasman commented on .github/workflows/gradle.yml:
```diff
@@ -0,0 +1,26 @@
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
```
Could we use a newer version of Java or are there some limitations?

AndreasNasman commented:
My main concern is the Java version. I think we should try to use the newest stable version if possible. According to [Wikipedia](https://en.wikipedia.org/wiki/Java_version_history), Java 17 will be LTS in September, and going from 11 to 17 feels like quite a big jump if we later decide to upgrade.

dusan-turajlic commented:
I agree! But I still think using the latest LTS is safer.

I can’t say if there are any pros or cons in using java 15 but let's evaluate if we should. A benefit would then be that we would have an easier time upgrading to 17.

But i guess the con is we need to deal with java 15 specific problems.

I’m not for or against either but let's check the current state and what others have experienced before merging.

Ether way great point to bring up 👍

dusan-turajlic commented:
> If you are making sure to use the very latest IDEs, frameworks and build tools and starting a greenfield project, you can, without hesitation, use Java 11 (LTS) or even the latest Java 15.

From [here](https://www.marcobehler.com/guides/a-guide-to-java-versions-and-features)

Based on the info there we don't realy gain anything by using Java 15 because we are using kotlin to write the code.

The Java release cycle has become alot more rapid, its now every 6 mounths so the jump from 11 to 17 is not as big as one might think.

But ether way It seems like we can totaly use 15. Maybe we should!

This is a Greengield project and why not use the latest stuff.

It's a risk either way, ether things don't work on 15 and we have to downgrade to 11 or we use 11 and upgrading to 17 is a pain!

IMO, let's just use 15.

AndreasNasman commented:
Yeah, version 15 sounds good! 👍

Since the backend will be written Kotlin, is the Java version specified here only for the CI process?

dusan-turajlic committed:
(baby-project) Upgrades the to java version 15

dusan-turajlic committed:
(baby-project) fixes comment to say 15

dusan-turajlic committed:
(baby-project) adds jenv .java-version to project

dusan-turajlic commented:
@AndreasNasman I updated the Java version and added the a `.java-version` to the project so if you use jenv it will automaticly pick up and use Java 15

dusan-turajlic committed:
(baby-project) adds inital test so the ci can run something

dusan-turajlic committed:
(baby-project) changes java version syntax in gradle.yml

dusan-turajlic committed:
(baby-project) uses a java 15 containr insted of extra steps

dusan-turajlic committed:
(baby-project) add the checkout to the correct branch

dusan-turajlic committed:
Adds stack trace for testing

dusan-turajlic committed:
WIP

dusan-turajlic committed:
maybe fixes the build

dusan-turajlic commented:
Note there is a downside with not running LTS (I relised now). Things like [JRebel](https://www.jrebel.com/products/jrebel) or [HotSwap](http://hotswapagent.org/) won't work with Java 15. But it would be good for longterm developement confert to use something like that. But lets merge this after the review is done so we can get started on the developemnt!

AndreasNasman reviewed (approved):
Looking good! 👍
